src/verify_cert.rs

```diff
@@ -14,14 +14,34 @@
 
 use crate::{
     cert::{self, Cert, EndEntityOrCa},
-    der, signed_data, subject_name, time, Error, SignatureAlgorithm, TrustAnchor,
+    der, signed_data, subject_name, time, CertRevocationList, Error, SignatureAlgorithm,
+    TrustAnchor,
 };
 
+/// A trait that can provide CRLs to use for revocation checking.
+pub trait CrlProvider<'a> {
+    /// A function that can be invoked with a [`Cert`] to optionally provide a [`CertRevocationList`]
+    /// to use to verify the certificate's revocation status.
+    ///
+    /// An implementation that only wishes to check revocation status for leaf certificates can
+    /// choose to return `None` when the [`Cert.ee_or_ca`] field is [`EndEntityOrCa::Ca`].
+    fn crl_for_cert(&self, cert: &Cert) -> Option<&'a CertRevocationList<'a>>;
```

Comment:
If this is the reason for making `Cert` part of the public API, perhaps we can restrict the fields of `Cert` that are made public to a smaller subset of fields that we think are necessary to determine which CRL applies? Presumably that would be a smallish subset.

Reply:
I agree (see my other comment).

I think we just need to send through:
* The serial (as `&[u8]`)
* The issuer (as `&[u8]`)
* An indication of whether the certificate is an end entity or intermediate (probably an enum made just for this purpose - the existing `EndEntityOrCa` enum exposes `Cert` in the `Ca` variant for the linked list)